Assets/Scripts/Model/Content/SecondEdition/Upgrades/Talent/SwiftApproach.cs

```diff
@@ -0,0 +1,74 @@
+﻿using Abilities.SecondEdition;
+using Actions;
+using ActionsList;
+using System;
+using System.Collections.Generic;
+using Upgrade;
+
+namespace UpgradesList.SecondEdition
+{
+    public class SwiftApproach : GenericUpgrade
+    {
+        public SwiftApproach()
+        {
+            UpgradeInfo = new UpgradeCardInfo(
+                "Swift Approach",
+                UpgradeType.Talent,
+                cost: 0,
+                abilityType: typeof(SwiftApproachAbility)
+            );
+
+            IsHidden = true;
+
+            IsWIP = true;
+        }
+    }
+}
+
+namespace Abilities.SecondEdition
+{
+    public class SwiftApproachAbility : GenericAbility
+    {
+        public override void ActivateAbility()
+        {
+            HostShip.OnBombWasDropped += CheckDroppedAbility;
+            HostShip.OnBombWasLaunched += CheckLaunchedAbility;
+        }
+
+        public override void DeactivateAbility()
+        {
+            HostShip.OnBombWasDropped -= CheckDroppedAbility;
+            HostShip.OnBombWasLaunched -= CheckLaunchedAbility;
+        }
+
+        private void CheckDroppedAbility()
+        {
+            if (Phases.CurrentPhase.Name != "Systems Phase") return;
+
+            RegisterAbilityTrigger(TriggerTypes.OnBombWasDropped, AskToPerformReposition);
```

Comment:
@Baledin Is this what you meant by splitting it so each method only registers one trigger?